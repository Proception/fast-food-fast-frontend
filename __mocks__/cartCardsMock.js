import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';
import { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder } from '../src/actions/orderActions';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [
      {
        "id": 1,
        "imageUrl": yam,
        "price": 500,
        "title": "Boiled Yam",
      },
      {
        "id": 2,
        "imageUrl": rice,
        "price": 900,
        "title": "Rice",
      },
      {
        "id": 3,
        "imageUrl": fufu,
        "price": 500,
        "title": "Fufu",
      },
    ],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: true,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};

const emptyCartProps = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: false,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};
export { props, emptyCartProps };
  