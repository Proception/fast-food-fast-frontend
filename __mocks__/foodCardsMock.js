import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: () => {},
  onChange: () => {}
};
export default props;
  