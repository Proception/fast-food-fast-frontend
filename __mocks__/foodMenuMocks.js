import { addItemToCart, updateCartTotal } from '../src/actions/orderActions';

const props = {
    addItemToCart,
    updateCartTotal,
    auth: {
      user: {
        id:1
      }
    }
    };
const emptyOrderProps = {
    updateCartTotal,
    addItemToCart,
    auth: {
    }
  };
export { props, emptyOrderProps };