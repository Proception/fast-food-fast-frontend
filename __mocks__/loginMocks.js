import { loginUser } from '../src/actions/authActions'

const props = {
    loginUser,
    auth: {
        isLoading: false,
    },
  };
  const propsIsLoading = {
    loginUser,
    auth: {
        isLoading: true,
    },
  };
  const state = {
    loginData: {
        email: 'omasan.esimaje',
        password: 'pass',
    }
  };
export { props, state, propsIsLoading };

