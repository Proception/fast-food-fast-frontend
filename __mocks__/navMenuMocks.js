import { setToken } from '../src/actions/authActions';

const stateGuest = {
    data: [],
    auth: {
        isAuthenticated: false,
    }
  };
  const stateUser = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
      }
    },
    setToken: setToken,
    order: {
      cart: [1,2]
    }
  };
  const stateUserEmptyCart = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
          role_id: 200
      }
    },
    setToken: setToken,
    order: {
      cart: []
    }
  };
  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZnVsbF9uYW1lIjoiQmVuZWRpY3QiLCJyb2xlX2lkIjoyMDAsImlhdCI6MTU1MTg3NzMyNSwiZXhwIjoxNTUxOTYzNzI1fQ.DwFFvZskGl9JeBnLD1CCnBSpYA1i0KgK1btUY1eLc9c'
  export { stateGuest, stateUser, token, stateUserEmptyCart };