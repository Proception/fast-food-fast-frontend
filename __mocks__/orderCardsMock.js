import { getOrders, getUserOrders } from '../src/actions/orderActions';
const props = {
    orders: [
        {
          "id": 1,
          "orderNo": "12",
          "price": 5000,
          "orderDate": "30 August, 2016",
          "orderStatus": "Pending",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 2,
          "orderNo": "54",
          "price": 1200,
          "orderDate": "Boiled Yam",
          "orderStatus": "In Progress",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 3,
          "orderNo": "67",
          "price": 1500,
          "orderDate": "30 October, 2017",
          "orderStatus": "Fulfilled",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 4,
          "orderNo": "33",
          "price": 3000,
          "orderDate": "30 January, 2018",
          "orderStatus": "Completed",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
      ],
    order: {
      orders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },],
      userOrders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },]
    },
    getOrders,
    getUserOrders
    };
const emptyOrderProps = {
  orders: [
      {
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 3,
        "orderNo": "67",
        "price": 1500,
        "orderDate": "30 October, 2017",
        "orderStatus": "Fulfilled",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 4,
        "orderNo": "33",
        "price": 3000,
        "orderDate": "30 January, 2018",
        "orderStatus": "Completed",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
    ],
  order: {
    orders: [],
    userOrders: [],
  },
  getOrders,
  getUserOrders
  };
export { props, emptyOrderProps };