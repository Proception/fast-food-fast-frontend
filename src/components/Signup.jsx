import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import {
  genericValidator, 
  emailValidator,
  passwordValidator,
  phoneNumberValidator
} from '../utils/CustomValidator';
import Input from './Forms/Input.jsx';
import Notify from '../utils/Notify';
import { makeSignupRequest, completeSignupRequest, signUpUser } from '../actions/authActions';

export class Signup extends Component {

  constructor(props) {
    super(props);

    this.state = {
      newUser: {
        name: '',
        email: '',
        pnumber: '',
        password: '',
        cpassword: '',
      },
    }
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleInput = this.handleInput.bind(this);
  }

  generateForms(){
    const InputData = [
      {
        type: 'email',
        name: 'email',
        value: this.state.newUser.email,
        placeholder: 'Email',
      },
      {
        type: 'text',
        name: 'name',
        value: this.state.newUser.name,
        placeholder: 'Full Name',
      },
      {
        type: 'number',
        name: 'pnumber',
        value: this.state.newUser.pnumber,
        placeholder: 'Phone Number',
      },
      {
        type: 'password',
        name: 'password',
        value: this.state.newUser.password,
        placeholder: 'Password',
      },
      {
        type: 'password',
        name: 'cpassword',
        value: this.state.newUser.cpassword,
        placeholder: 'Confirm Password',
      },
    ];

    return InputData.map(
      (item, index) => <p><Input
        key={index} type={ item.type } name ={ item.name }
        value={ item.value } placeholder= { item.placeholder }
        handleChange={ this.handleInput } myClass = {'reg-control'}/></p>,
    );
  }

  handleInput(e) {
    let value = e.target.value;
    let name = e.target.name;
    this.setState(prevState => {
      return {
        newUser: {
          ...prevState.newUser, [name]: value
        }
      }
    }, () => {}
    )
  }

  checkInput(userData){
    Object.entries(userData).forEach(
      ([key, value]) => {
        if (!genericValidator(value)) {
          if (key === 'cpassword') {
            Notify.notifyError('Please enter a confirm password');
            throw BreakException;
          }
          if (key === 'pnumber') {
            Notify.notifyError('Please enter a Phone Number');
            throw BreakException;
          }
          Notify.notifyError(`Please enter a valid ${key}`);
          throw BreakException;
        }
      }
    );
  }

  validateInput(email, pnumber, password, cpassword){
    if (!emailValidator(email)) {
      Notify.notifyError(`${email} is not a valid email`);
      throw BreakException;
    }
    if (!phoneNumberValidator(pnumber)) {
      Notify.notifyError(`${pnumber} is not a phone number`);
      throw BreakException;
    } 

    if (!passwordValidator(password, cpassword)) {
      Notify.notifyError('Passwords do not match');
      throw BreakException;
    }
  }

  handleFormSubmit(e) {
    e.preventDefault();
    let userData = this.state.newUser;

    const { 
      email,
      pnumber,
      password,
      cpassword
    } = userData;

    try {
      this.checkInput(userData);
      this.validateInput(email, pnumber, password, cpassword);
      this.props.signUpUser(userData);
    } catch (error) {
      
    }
  }

  render() {
    return (
      <section>
        <div class="container">
          <div id="wrapper">
            <h1>Enter Registration Details</h1>
            <div class="register">
              <div id="notif"></div>
              <form id="signupForm" noValidate onSubmit={this.handleFormSubmit}>
                {
                  this.generateForms()
                }
                <p><Input type={'submit'} disabled={this.props.auth.isLoading}
                  value={this.props.auth.isLoading ? 'Processing' : 'submit'}
                  myClass={'button'}
                /></p>
                <h5>Already have an account click <NavLink to="/login" >here</NavLink></h5>
              </form>
            </div>
          </div>
        </div>
      </section>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { makeSignupRequest, completeSignupRequest, signUpUser })(Signup);