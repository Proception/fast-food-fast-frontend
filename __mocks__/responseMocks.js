import { token } from './navMenuMocks';

export const okResponse = {
    status: 200,
    response: {
        code: 200,
        status: 'ok',
        messages: 'User logged in successfully',
        data: token
    }
};

export const unauthorizedResponse = {
    status: 401,
    response: {
        code: 401,
        status: 'ok',
        messages: 'User is not authorized',
        data: token
    }
};

export const createdResponse = {
    status: 201,
    response: {
        code: 201,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const noContentResponse = {
    status: 204,
    response: {
        code: 204,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const conflictResponse = {
    status: 409,
    response: {
        code: 409,
        status: 'Conflict',
        messages: 'User exists on the platform',
        data: token
    }
}

export const badRequestResponse = {
    status: 400,
    response: {
        code: 400,
        status: 'Bad Request',
        messages: 'Invalid credentials',
        data: token
    }
}
