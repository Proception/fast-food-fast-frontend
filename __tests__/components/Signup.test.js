import React from 'react';
import { Signup } from '../../src/components/Signup.jsx';
import { props, state, propsIsLoading } from '../../__mocks__/loginMocks';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Signup state={ state } {...props} />);
});
describe('Signup Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render signup with isLoading true', () => {
    const  wrapper = shallow(<Signup state={ state } {...propsIsLoading} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls handleFormSubmit(e)', () => {
    const e = {
      preventDefault: ()=>{}
    }
    
    sinon.spy(wrapper.instance(), 'handleFormSubmit');
    wrapper.instance().handleFormSubmit(e);
    expect(wrapper.instance().handleFormSubmit.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().handleFormSubmit.calledWith(e));
  });
  it('calls handleInput(e)', () => {
    const e1 = {
      preventDefault: ()=>{},
      target: {
        name: "email",
        value: "omasan.esimaje@gmail.com"
      },
    }
    
    sinon.spy(wrapper.instance(), 'handleInput');
    wrapper.instance().handleInput(e1);
    expect(wrapper.instance().handleInput.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().handleInput.calledWith(e1));
  });
  it('calls handleInput(e) without validations', () => {
    const data = [
      { 
        name: 'Ben',
      },
      {
        email: 'ben@gmail.com',
      },
      {
        pnumber: '07062257273',
      },
      {

        password: 'pass',
      },
      {
        cpassword: 'pass',
      },
    ]
    
    sinon.spy(wrapper.instance(), 'handleInput');
    sinon.spy(wrapper.instance(), 'handleFormSubmit');

    data.forEach(e => {
      const e1 = {
        preventDefault: e.preventDefault,
        target: {
          name: Object.keys(e)[0],
          value: Object.values(e)[0]
        },
      }
      const event = {
        preventDefault: ()=>{}
      }
      wrapper.instance().handleInput(e1);
      wrapper.instance().handleFormSubmit(event);
      expect(wrapper.instance().handleFormSubmit.calledWith(event));
    });
    expect(wrapper.instance().handleInput.callCount)
      .toEqual(5);
  });
  it('calls handleInput(e) with email validations', () => {
    const data = [
      { 
        name: 'Ben',
      },
      {
        email: 'bengmail.com',
      },
      {
        pnumber: '07062257273',
      },
      {

        password: 'pass',
      },
      {
        cpassword: 'pass',
      },
    ]
    
    sinon.spy(wrapper.instance(), 'handleInput');
    sinon.spy(wrapper.instance(), 'handleFormSubmit');

    data.forEach(e => {
      const e1 = {
        preventDefault: e.preventDefault,
        target: {
          name: Object.keys(e)[0],
          value: Object.values(e)[0]
        },
      }
      const event = {
        preventDefault: ()=>{}
      }
      wrapper.instance().handleInput(e1);
      wrapper.instance().handleFormSubmit(event);
      expect(wrapper.instance().handleFormSubmit.calledWith(event));
    });
    expect(wrapper.instance().handleInput.callCount)
      .toEqual(5);
  });
  it('calls handleInput(e) with phone validations', () => {
    const data = [
      { 
        name: 'Ben',
      },
      {
        email: 'ben@gmail.com',
      },
      {
        pnumber: 'ererererr',
      },
      {

        password: 'pass',
      },
      {
        cpassword: 'pass',
      },
    ]
    
    sinon.spy(wrapper.instance(), 'handleInput');
    sinon.spy(wrapper.instance(), 'handleFormSubmit');

    data.forEach(e => {
      const e1 = {
        preventDefault: e.preventDefault,
        target: {
          name: Object.keys(e)[0],
          value: Object.values(e)[0]
        },
      }
      const event = {
        preventDefault: ()=>{}
      }
      wrapper.instance().handleInput(e1);
      wrapper.instance().handleFormSubmit(event);
      expect(wrapper.instance().handleFormSubmit.calledWith(event));
    });
    expect(wrapper.instance().handleInput.callCount)
      .toEqual(5);
  });
  it('calls handleInput(e) with password validations', () => {
    const data = [
      { 
        name: 'Ben',
      },
      {
        email: 'ben@gmail.com',
      },
      {
        pnumber: '07062257273',
      },
      {

        password: 'pass',
      },
      {
        cpassword: 'passs',
      },
    ]
    
    sinon.spy(wrapper.instance(), 'handleInput');
    sinon.spy(wrapper.instance(), 'handleFormSubmit');

    data.forEach(e => {
      const e1 = {
        preventDefault: e.preventDefault,
        target: {
          name: Object.keys(e)[0],
          value: Object.values(e)[0]
        },
      }
      const event = {
        preventDefault: ()=>{}
      }
      wrapper.instance().handleInput(e1);
      wrapper.instance().handleFormSubmit(event);
      expect(wrapper.instance().handleFormSubmit.calledWith(event));
    });
    expect(wrapper.instance().handleInput.callCount)
      .toEqual(5);
  });
});
