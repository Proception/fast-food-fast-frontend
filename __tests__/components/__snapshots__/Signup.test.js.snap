// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component should match snapshot 1`] = `
<section>
  <div
    class="container"
  >
    <div
      id="wrapper"
    >
      <h1>
        Enter Registration Details
      </h1>
      <div
        class="register"
      >
        <div
          id="notif"
        />
        <form
          id="signupForm"
          noValidate={true}
          onSubmit={[Function]}
        >
          <p>
            <Input
              handleChange={[Function]}
              key="0"
              myClass="reg-control"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="1"
              myClass="reg-control"
              name="name"
              placeholder="Full Name"
              type="text"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="2"
              myClass="reg-control"
              name="pnumber"
              placeholder="Phone Number"
              type="number"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="3"
              myClass="reg-control"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="4"
              myClass="reg-control"
              name="cpassword"
              placeholder="Confirm Password"
              type="password"
              value=""
            />
          </p>
          <p>
            <Input
              disabled={false}
              myClass="button"
              type="submit"
              value="submit"
            />
          </p>
          <h5>
            Already have an account click 
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/login"
            >
              here
            </NavLink>
          </h5>
        </form>
      </div>
    </div>
  </div>
</section>
`;

exports[`Signup Component should render signup with isLoading true 1`] = `
<section>
  <div
    class="container"
  >
    <div
      id="wrapper"
    >
      <h1>
        Enter Registration Details
      </h1>
      <div
        class="register"
      >
        <div
          id="notif"
        />
        <form
          id="signupForm"
          noValidate={true}
          onSubmit={[Function]}
        >
          <p>
            <Input
              handleChange={[Function]}
              key="0"
              myClass="reg-control"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="1"
              myClass="reg-control"
              name="name"
              placeholder="Full Name"
              type="text"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="2"
              myClass="reg-control"
              name="pnumber"
              placeholder="Phone Number"
              type="number"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="3"
              myClass="reg-control"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </p>
          <p>
            <Input
              handleChange={[Function]}
              key="4"
              myClass="reg-control"
              name="cpassword"
              placeholder="Confirm Password"
              type="password"
              value=""
            />
          </p>
          <p>
            <Input
              disabled={true}
              myClass="button"
              type="submit"
              value="Processing"
            />
          </p>
          <h5>
            Already have an account click 
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/login"
            >
              here
            </NavLink>
          </h5>
        </form>
      </div>
    </div>
  </div>
</section>
`;
